src/main/java/com/iota/iri/network/pipeline/TransactionProcessingPipelineImpl.java

```diff
@@ -135,7 +143,7 @@ private void addStage(String name, BlockingQueue<ProcessingContext> queue,
                             receivedStageQueue.put(payload.getRight());
                             break;
                         case BROADCAST:
-                            broadcastStageQueue.put(ctx);
+                            broadcastStageQueue.add(ctx);
```

Comment:
@galrogo Just to address this comment and subsequently issue #1512, currently transaction requests are dependent on broadcasting, so if you do not broadcast unsolid transactions, you will never send a transaction request. So it is necessary to pass on transactions before they are solid in the current framework. May be worth discussing a new method of transaction requesting from neighbours to avoid this dependency.

Reply:
Can we just broadcast random solid tips like before?

I guess moving on to the new messages that were developed for hornet will solve this problem

Reply:
IIUC, you implemented my suggestion above?

Reply:
When you say before are you referring to before the refactor? Because as is, after a transaction is placed into the solidifier, the solidify stage will fetch a random solid tip and broadcast it. If the tip is not solid the pipeline goes straight to `FINISH` instead. As it was before it was: 
`Transaction -> Pre-Processing -> Hashing/Validation -> Received (store) -> Broadcast (pass transaction on before verification of solidity) -> Finish`
But now it would be: 
`Transaction -> Pre-Processing -> Hashing/Validation -> Received (store) -> Solidify (check if solid, if so pass on to broadcast, if not, place in solidification queue and fetch random tip for broadcasting) -> Broadcast -> Finish`

Reply:
@DyrellC, marking this as point to discuss before merge

Reply:
never mind, I am good